Document Profiling+SSI beta k8s support (#25237)
DataDog/documentation: content/en/profiler/enabling/ssi.md

  image:
    tag: latest
datadog:
  profiling:
    enabled: auto
```

If you already have Datadog Helm Chart, ensure it is updated to at least version 3.71.0.
Image tag "latest" is used to install a recent agent version with support for profiling as Datadog
Helm Chart defaults to an older agent version.

[2]: https://app.datadoghq.com/account/settings/agent/latest?platform=overview
{{% /tab %}}
{{% tab "Kubernetes with Datadog Operator" %}}

1. Go to the [Agent Installation Page][2] and select Kubernetes, then select Operator.
1. Open the APM dropdown and toggle the Enable APM Instrumentation switch.
1. Add the values below to `datadog-values.yaml` in addition to those indicated by the installation page. `DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED=auto` environment variable on
the cluster agent turns on automatic profiler enablement for any new process worth profiling.
1. Proceed with rest of the installation instructions.

```
spec:
  override:
    nodeAgent:
      image:
        tag: latest
    clusterAgent:
      image:
        tag: latest
      env:
        - name: DD_ADMISSION_CONTROLLER_AUTO_INSTRUMENTATION_PROFILING_ENABLED
          value: "auto"
```

> Image tag "latest" is used to install a recent agent version with support for profiling as Datadog
> Operator defaults to an older agent version.

Reviewer: ```suggestion
Use the "latest" image tag to install a recent Agent version with support for profiling, as 
the Datadog Operator defaults to an older Agent version.
```

[2]: https://app.datadoghq.com/account/settings/agent/latest?platform=overview
{{% /tab %}}
{{< /tabs >}}

## How does profiling work with SSI

After the installation, all new processes are executed with the `DD_PROFILING_ENABLED=auto`

> environment variable. Running processes will not be affected. The Datadog library dynamically turns

Reviewer: ```suggestion
environment variable. Running processes are not affected. The Datadog library dynamically turns
```

on the profiler for the processes that are good profiling candidates.

> The logic for identifying a process as a good candidate varies by language. For Java, all processes
> are profiled as Java applications are usually deployed as a single Java process on a host. For Node

Reviewer: ```suggestion
The logic for identifying a process as a good candidate varies by language. For Java, all processes
are profiled, as Java applications are usually deployed as a single Java process on a host. For Node
```

and Python, profiler is only turned on if the application is running for more than 30 seconds and
has created at least one tracing span.

SSI can also be configured to inject profiling on each and every process by using the value `true`
instead of `auto`.

**Note**: Datadog recommends using the `auto` setting to avoid profiling low value processes.

# Reverting

The [Single Step APM Instrumentation][1] page contains instructions for removing all
instrumentation. Removing instrumentation also removes profiling.

> Additionally, you can disable profiling by:

Reviewer: ```suggestion
Additionally, you can disable profiling by taking one of the following steps:
```

> * repeating the installation instructions using `false` value in place of `auto`, or
> * removing the `DD_PROFILING_ENABLED` setting from the `/etc/environment` file on the host, for
>   host and container deployments.

Reviewer: ```suggestion
* Repeat the installation instructions using a `false` value in place of `auto`.
* Remove the `DD_PROFILING_ENABLED` setting from the `/etc/environment` file on the host, for host and container deployments.
```
